### DISCUSSION TITLE
need_part multiple lines support

### DISCUSSION THREAD

**Discussion by jumbo172** [useblocks/sphinx-needs#1283] [General]
I am using sphinx-needs 2.1.0. Can `:need_part:` support multiple lines of text? I have a paragraph that I need to reference in different needs.

```
   :need_part:`(1) some stuff on a single line works`

   :need_part:`(1) some more stuff
   using multiple lines does not work`
```

**Comment by chrisjsewell** [comment]
`need_part` is just a standard restructuredtext role (a.k.a intepreted text), it gets parsed the same as every other role, and so yes it can (from docutils CLI):

```console
$ echo ":strong:\`hi\nthere\`" | rst2pseudoxml
<document source="<stdin>">
    <paragraph>
        <strong>
            hi
            there
```

**Reply by jumbo172** [reply]
Hi Chris,

Ok, so it supports multiple lines in the rst, but it does not render them as multiple lines in the HTML:

.. test_case:: example1

   :need_part:`(1A) some stuff
   does not render anything after the word stuff`

.. test_case:: example2

   :need_part:`(2A) some stuff\n
   does not render anything after the word stuff`

.. test_case:: example3

   :need_part:`(3A) | some stuff
   does not render anything after the word stuff`

.. test_case:: example4

   :need_part:`(4A) some stuff \
   does render it all, but all on a single line when the rst uses 2 lines` 

.. test_case:: example5

   `I want the need_part to encompass everything in the ticks and render like a normal need does.

   So I have a new line without any special formatting.

   And a list:

   * one 
   * two`

Renders as:

![need_paragrahp2](https://github.com/user-attachments/assets/9f5c2fff-682e-419f-a87c-15211adf3969)

**Reply by chrisjsewell** [reply]
fixed https://github.com/useblocks/sphinx-needs/pull/1284
